### PR TITLE
Add `set_sample_data` keys and explain how they work

### PR DESCRIPTION
**Pull request by matsimitsu**
@jeffkreeftmeijer are my assumptions correct on this call on the Elixir side? 

@tombruijn Am I correct, that since `sample_data` method here is called after the transaction is closed, only `set_sample_data("custom_data", {})` will work? ( https://github.com/appsignal/appsignal-ruby/blob/master/lib/appsignal/transaction.rb#L105 ) 

@xadamy Are the examples for NodeJS correct? The docs seem to suggest only string values are accepted, not objects/arrays? Do the customers have to jsonfiy the string themselves?

### CODE REVIEW

**Comment by xadamy** [comment]
@matsimitsu Which page did you see this? [This page](https://docs.appsignal.com/nodejs/instrumentation/span.html) lists the method signature as the following: `span.setSampleData(key: string, data: Array<any> | { [key: string]: any })`.

**Comment by matsimitsu** [comment]
> @matsimitsu Which page did you see this? This page lists the method signature as the following: span.setSampleData(key: string, data: Array<any> | { [key: string]: any }).

@xadamy that's the page :) Maybe I don't understand TS types enough yet :P

File: source/nodejs/instrumentation/span.html.md [hunk 1]
```diff
@@ -55,23 +55,76 @@ span
...
 
-### `span.setSampleData(key: string, data: Array<any> | { [key: string]: any })`
+## `span.setSampleData(key: string, data: Array<any> | { [key: string]: any })`
 
-Adds sample data to the current `Span`. The sample data object must not contain any nested objects.
+Adds sample data to the current `Span`. The sample data object must not contain any nested objects. This call accepts the following keys, some keys require a specific format.
 
-### `span.close(endTime?: number)`
+
+### `sesssion_data`
```
Comment (xadamy) [inline source/nodejs/instrumentation/span.html.md:75]:
```suggestion
### `session_data`
```

File: source/nodejs/instrumentation/span.html.md [hunk 2]
```diff
@@ -55,23 +55,76 @@ span
...
+```
+
+This key accepts nested objects that will be rendered as JSON on a Incident Sample page for both Exception and Performance samples.
+
+![sesssion_data](/assets/images/screenshots/sample_data/session_data.png)
+
+
+
+### `params`
+Filled with framework (such as Phoenix) parameters by default, but can be overridden or filled with the following call:
```
Comment (xadamy) [inline source/nodejs/instrumentation/span.html.md:90]:
```suggestion
Filled with framework (such as Express) parameters by default, but can be overridden or filled with the following call:
```

File: source/nodejs/instrumentation/span.html.md [hunk 3]
```diff
@@ -55,23 +55,76 @@ span
...
+
+```
+span.setSampleData("params", {action: "show", controller: "homepage"})
```
Comment (xadamy) [inline source/nodejs/instrumentation/span.html.md:93]:
```suggestion
span.setSampleData("params", {string: "value", number: 123 })
```

File: source/nodejs/instrumentation/span.html.md [hunk 4]
```diff
@@ -55,23 +55,76 @@ span
...
+```
+
+This key accepts nested objects and will show up as follows on a Incident Sample page for both Exception and Performance samples, formatted as JSON.
+
+![params](/assets/images/screenshots/sample_data/params.png)
+
+
+
+### `environment`
+Environment variables from a request/background job, filled by the Phoenix integration, but can be filled/overriden with the following call:
```
Comment (xadamy) [inline source/nodejs/instrumentation/span.html.md:103]:
```suggestion
Environment variables from a request/background job (typically filled by the default `http` integration, but can be further augmented by other integrations), but can be filled/overriden with the following call:
```

File: source/nodejs/instrumentation/span.html.md [hunk 5]
```diff
@@ -55,23 +55,76 @@ span
...
 
 Sets the `namespace` of the current `Span`. Namespaces are a way to group error incidents, performance incidents, and host metrics in your app.
 
-### `span.addError(error: Error | object)`
+## `span.addError(error: Error | object)`
 
 Sets the `error` of the current `Span`. When an `Error` object is passed to the `setError` method, the `stack` property is normalised and transformed into an array of strings, with each string representing a line in the stacktrace.
 
-### `span.setSampleData(key: string, data: Array<any> | { [key: string]: any })`
+## `span.setSampleData(key: string, data: Array<any> | { [key: string]: any })`
```
Comment (xadamy) [inline source/nodejs/instrumentation/span.html.md:70]:
```suggestion
## `span.setSampleData(key: string, data: Array<string | number | boolean> | { [key: string]: string | number | boolean })`
```

Fixing my own work here!

File: source/ruby/instrumentation/tagging.html.md [hunk 6]
```diff
@@ -50,7 +50,23 @@ Appsignal.tag_request(
...
 ```
 
-## Link templates
+### Link templates
 
 Tags can also be used to create link templates. Read more about link templates
 in our [link templates guide](/application/link-templates.html).
+
+
+## Sample Data
```
Comment (tombruijn) [inline source/ruby/instrumentation/tagging.html.md:59]: Should this page for Ruby also be renamed to "Tagging and Sample Data"?

**Pull request merged**